disco_aws_automation/disco_group.py

```diff
@@ -143,6 +163,19 @@ def update_group(self, hostclass, desired_size=None, min_size=None, max_size=Non
         """
         Create a new autoscaling group or update an existing one
         """
+        existing_group = self.get_existing_group(hostclass, group_name)
+
+        # if there is an existing group and create_if_exists is false then we will be updating that group
+        # in that case, use the group type of the existing group instead of the passed in type
```

Comment:
But if the existing group is type `asg`, the new group will also be of type `asg`, instead of `spot` that we want it to be.